lib/responses/errors.go

```diff
@@ -58,3 +59,32 @@ func HTTPErrorHandler(err error, c echo.Context) {
 	}
 	// TODO: use an error matching the error code
 }
+
+// this is a simple way to try to convert err.Message interface
+// to ErrorResponse without external packages
+func errToErrorResponse(err error) *ErrorResponse {
```

Comment:
since this is map, i think converting to json and back to struct is the only built in way in golang to do this
there is package for decoding struct, but i did not want to add it just for this